src/datachain/lib/file.py

```diff
@@ -242,6 +242,30 @@ def __init__(self, **kwargs):
         self._catalog = None
         self._caching_enabled: bool = False
 
+    def as_text_file(self) -> "TextFile":
+        """Convert the file to a `TextFile` object."""
+        if isinstance(self, TextFile):
+            return self
+        file = TextFile(**self.model_dump())
+        file._set_stream(self._catalog, caching_enabled=self._caching_enabled)
+        return file
+
+    def as_image_file(self) -> "ImageFile":
+        """Convert the file to a `ImageFile` object."""
+        if isinstance(self, ImageFile):
+            return self
+        file = ImageFile(**self.model_dump())
```

Comment:
Same here, we should probably look into proper handling of error if the file cannot be converted to image file or if it is an invalid image file.

Reply:
User will get an error in case they try to use methods from `ImageFile` class on non-image file. Here is just a matter of model type, not real file.